agent/tests/TestCase.php

```diff
@@ -113,17 +117,31 @@ protected function functionName(): string
         return static::class.'::'.debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, limit: 2)[1]['function'];
     }
 
-    protected function assertLogMatches(string $expected, string $actual, bool $silent = false, bool $quiet = false): self
+    protected function assertLogMatches(string $expected, string $actual, bool $silent = false, bool $quiet = false, bool $verbose = false): self
     {
         if (! $quiet && ! $silent) {
             $expected = "{date} {info} Nightwatch agent initiated: Listening on \[127.0.0.1:\d{4}\]\n{$expected}";
         }
+
+        if ($verbose) {
+            $expectedSignature = trim(self::getSignature());
+            $expectedSignaturePath = __DIR__.'/../build/signature.txt';
+
+            $expected = "{date} {debug} Read signature from \[{$expectedSignaturePath}\] and found \[{$expectedSignature}\]\n{$expected}";
+        }
+
         $expected = str_replace('{date}', '\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}', $expected);
         $expected = str_replace('{duration}', '\[\d(\.\d{1,3})?s\]', $expected);
         $expected = str_replace('{info}', '\[INFO\]', $expected);
         $expected = str_replace('{error}', '\[ERROR\]', $expected);
+        $expected = str_replace('{debug}', '\[DEBUG\]', $expected);
 
-        $this->assertMatchesRegularExpression("#^{$expected}$#", $actual);
+        $expected = explode(PHP_EOL, $expected);
+        $actual = explode(PHP_EOL, $actual);
+
+        foreach ($expected as $index => $e) {
+            $this->assertMatchesRegularExpression("#^{$e}$#", $actual[$index]);
+        }
```

Comment:
Made a tweak to this so that it compares log line by line.

This makes the error message much more useful, as it points you to the problematic log line. It can be hard to determine which line is problematic when it outputs a massive regex string.